src/App.jsx

```diff
@@ -1,36 +1,89 @@
-import React, { useEffect, useState } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 import Body from "./Layouts/body/Body";
 import Footer from "./Layouts/footer/Footer";
 import Header from "./Layouts/header/Header";
 import ErrorBoundary from "./Components/ErrorBoundary";
-import { api } from './API/PostService'
+import { api } from './API/PostService';
 
-import VidoeModal from './Components/UI/videomodal/VideoModal'
-import VideoForm from './Components/UI/videoform/VideoForm'
+import Modal from './Components/UI/modal/Modal';
+import VideoFormModal from './Components/UI/videoformmodal/VideoFormModal';
+import VideoDetails from "./Components/VideoDetails";
+import MyButton from './Components/UI/button/MyButton';
+import useModalState from './hooks/useModalState';
 
 const App = () => {
 
   const [videos, setVideos] = useState([]);
+  const [isVideoDetail, setIsVideoDetail] = useState(false);
+  const [videoDetaill, setVideoDetaill] = useState();
+
+  const modal = useModalState();
 
   useEffect(() => {
     setVideos(api);
   }, []);
 
   const createVideo = (newVideo) => {
+    newVideo.id = videos.length+1;
+
     setVideos([...videos, newVideo]);
+  };
+
+  const videoDetail = (id) => {
+    console.log("render");
+    setVideoDetaill(videos[id - 1]);
+    setIsVideoDetail(true);
+  };
+
+  const editVideo = (video) => {
+    const editedVideos = videos.map(current => {
+      return current.id === video.id ? video : current;
+    });
+
+    setVideos(editedVideos);
   }
 
-  const [modal, setModal] = useState(false);
+  const removeVideo = (video) => {
+    setVideos(videos.filter(v => v.id !== video.id));
+  };
 
   return(
     <>
       <ErrorBoundary>
-        <VidoeModal className='videoModal' title='add movie' visible={modal} setVisible={setModal}>
-          <VideoForm creat={createVideo} showModal={setModal} />
-        </VidoeModal>
+        {modal.isAddType &&
+         <Modal className='videoModal' title='add movie' onModalClose={modal.close}>
+           <VideoFormModal onSubmit={createVideo} hideModal={modal.close} />
+         </Modal>
```

Comment:
I would still recommend to do approach I mentioned in https://github.com/StepchUk/reactgmp22/pull/4#discussion_r898998701, but it's not necessary